core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/InProcessTelemetryChannel.java

```diff
@@ -47,313 +45,339 @@
+
+	private static AtomicLong itemsSent = new AtomicLong(0);
+
+	public InProcessTelemetryChannel() {
+		boolean developerMode = false;
+		try {
+			String developerModeAsString = System.getProperty(DEVELOPER_MODE_SYSTEM_PROPRETY_NAME);
+			if (!LocalStringsUtils.isNullOrEmpty(developerModeAsString)) {
+				developerMode = Boolean.valueOf(developerModeAsString);
+			}
+		} catch (Throwable t) {
+			developerMode = false;
+			InternalLogger.INSTANCE.trace("%s generated exception in parsing," + "stack trace is %s",
+					DEVELOPER_MODE_SYSTEM_PROPRETY_NAME, ExceptionUtils.getStackTrace(t));
+		}
+		initialize(null, null, developerMode, createDefaultMaxTelemetryBufferCapacityEnforcer(null),
+				createDefaultSendIntervalInSecondsEnforcer(null), true);
+	}
+
+	/**
+	 * Ctor
+	 *
+	 * @param endpointAddress
+	 *            Must be empty string or a valid uri, else an exception will be
+	 *            thrown
+	 * @param developerMode
+	 *            True will behave in a 'non-production' mode to ease the debugging
+	 * @param maxTelemetryBufferCapacity
+	 *            Max number of Telemetries we keep in the buffer, when reached we
+	 *            will send the buffer Note, value should be between
+	 *            TRANSMIT_BUFFER_MIN_TIMEOUT_IN_MILLIS and
+	 *            TRANSMIT_BUFFER_MAX_TIMEOUT_IN_MILLIS inclusive
+	 * @param sendIntervalInMillis
+	 *            The maximum number of milliseconds to wait before we send the
+	 *            buffer Note, value should be between
+	 *            MIN_MAX_TELEMETRY_BUFFER_CAPACITY and
+	 *            MAX_MAX_TELEMETRY_BUFFER_CAPACITY inclusive
+	 */
+	public InProcessTelemetryChannel(String endpointAddress, boolean developerMode, int maxTelemetryBufferCapacity,
+			int sendIntervalInMillis) {
+		initialize(endpointAddress, null, developerMode,
+				createDefaultMaxTelemetryBufferCapacityEnforcer(maxTelemetryBufferCapacity),
+				createDefaultSendIntervalInSecondsEnforcer(sendIntervalInMillis), true);
+	}
+
+	/**
+	 * This Ctor will query the 'namesAndValues' map for data to initialize itself
+	 * It will ignore data that is not of its interest, this Ctor is useful for
+	 * building an instance from configuration
+	 *
+	 * @param namesAndValues
+	 *            - The data passed as name and value pairs
+	 */
+	public InProcessTelemetryChannel(Map<String, String> namesAndValues) {
+		boolean developerMode = false;
+		String endpointAddress = null;
+		int maxInstantRetries = DEFAULT_MAX_INSTANT_RETRY;
+		
+		LimitsEnforcer maxTelemetryBufferCapacityEnforcer = createDefaultMaxTelemetryBufferCapacityEnforcer(null);
+
+		LimitsEnforcer sendIntervalInSecondsEnforcer = createDefaultSendIntervalInSecondsEnforcer(null);
+
+		boolean throttling = true;
+		if (namesAndValues != null) {
+			throttling = Boolean.valueOf(namesAndValues.get("Throttling"));
+			developerMode = Boolean.valueOf(namesAndValues.get(DEVELOPER_MODE_NAME));
+			try {
+				maxInstantRetries = Integer.parseInt(INSTANT_RETRY_NAME);
```

Comment:
I think this will blow. Shouldn't it be Integer.parseInt(namesAndValues.get(INSTANT_RETRY_NAME)); ?

Reply:
Yes good catch. This will blow

Reply:
Yes very good catch. Fixed and added null check.

Reply:
Integer.parseInt does throw NumberFormatException which I already had a catch for. It should only log if we have ERROR tracing on.